frontend/lib/src/controllers/user_profile_controller.dart

```diff
@@ -0,0 +1,53 @@
+import 'package:get/get.dart';
+
+import 'package:frontend/src/models/user.dart';
+import 'package:frontend/src/services/user_profile_service.dart';
+
+class UserProfileController extends GetxController {
+  final RxBool loading = true.obs;
+  final RxBool _buttonEnabled = true.obs;
+  final RxString _userName = ''.obs;
```

Comment:
Change this implementation from `RxString` to an observable `User` model

https://www.youtube.com/watch?v=NrtIfLPleBY